pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicsConsumerImplTest.java

```diff
@@ -1170,6 +1193,20 @@ public void testGetLastMessageId() throws Exception {
             }
         });
 
+        msgIds = consumer.getLastMessageIds();
+        assertEquals(msgIds.size(), 6);
+        assertEquals(msgIds.stream().map(TopicMessageId::getOwnerTopic).collect(Collectors.toSet()), topics);
+        for (TopicMessageId msgId : msgIds) {
+            int numMessages = (int) ((MessageIdAdv) msgId).getEntryId() + 1;
+            if (msgId.getOwnerTopic().equals(topicName1)) {
+                assertEquals(numMessages, totalMessages * 2);
+            } else if (msgId.getOwnerTopic().startsWith(topicName2)) {
+                assertEquals(numMessages, totalMessages / 2 * 2);
```

Comment:
```suggestion
                assertEquals(numMessages, totalMessages);
```

It is best to keep the same way as before:

https://github.com/apache/pulsar/pull/20040/files#diff-dc5cb0bbc25d4856fb9e07ace3e45dc27f4da5ac208cf4ef9a885e08416a9eb9R1187-R1192